pkg/telemetry/closeable_metric_sink_adapter.go

```diff
@@ -0,0 +1,45 @@
+package telemetry
+
+import "github.com/armon/go-metrics"
+
+type CloseableMetricSinkAdapter struct {
+	delegate metrics.MetricSink
+}
+
+var _ CloseableMetricSink = (*CloseableMetricSinkAdapter)(nil)
```

Comment:
This struct just allows us to take a plain ol' MetricSink and slap a Closeable() method onto it, in effect. In practice it's used to adapt the built-in BlackholeSink which gets used when no telemetry socket is specified.